### ISSUE TITLE
Logic needs timing control on fast processors ...

### ISSUE THREAD

**Issue by nkolban** [wayoda/LedControl#11]
We have a project underway to run Sketches written for Arduinos on the Raspberry Pi.   With its GHz processor, it is 50x faster than the Arduino.  This is causing problems with the library which is bit banging SPI because the timings of the bits now become 50x faster.  Obviously the library is perfect for real arduinos but we might want to address the notion that CPU speeds will get faster and logic may break.  For example, I believe that max clock for the 7219 is 10MHz ... but at 1GHz ... processor speeds and bit banging ... the time which a signal is high or low may be too short without some timing logic built in or hardware SPI timing controls.

**Comment by gordoste** [comment]
Yes, I ran into this issue when running MAX7219 from ESP8266 with 2 metres of wire. The frequency of the signals is high enough that the signal cannot be received properly by the MAX7219. Running same code on Arduino Uno works perfectly but I want WiFi :)
